### PR TITLE
Rename HttpFilters to HttpInterceptors

### PR DESCRIPTION
**Pull request by rhyshort**
Rename HttpFilters to HttpInterceptors as per slack discussion. This is to Remove confusing between this API and the changes filter API.

reviewer @tomblench 
reviewer @mikerhodes

### CODE REVIEW

File: sync-core/src/main/java/com/cloudant/http/HttpConnection.java [hunk 1]
```diff
@@ -200,10 +200,10 @@ public HttpConnection execute() throws IOException {
                     connection.setRequestProperty("Content-type", contentType);
                 }
 
-                HttpConnectionFilterContext currentContext = new HttpConnectionFilterContext(this);
+                HttpConnectionInterceptorContext currentContext = new HttpConnectionInterceptorContext(this);
 
-                for (HttpConnectionRequestFilter requestFilter : requestFilters) {
-                    currentContext = requestFilter.filterRequest(currentContext);
+                for (HttpConnectionRequestInterceptor requestFilter : requestInterceptors) {
+                    currentContext = requestFilter.interceptRequest(currentContext);
```
Comment (mikerhodes) [inline sync-core/src/main/java/com/cloudant/http/HttpConnection.java]: requestFilter var name

Comment (rhyshort) [inline sync-core/src/main/java/com/cloudant/http/HttpConnection.java]: done in 108e3ad

File: sync-core/src/main/java/com/cloudant/http/CookieInterceptor.java [hunk 2]
```diff
@@ -56,13 +56,13 @@
      * @param username The username to use when getting the cookie
```
Comment (mikerhodes) [inline sync-core/src/main/java/com/cloudant/http/CookieInterceptor.java]: Line 55 has "Constructs a cookie filter".

Comment (mikerhodes) [inline sync-core/src/main/java/com/cloudant/http/CookieInterceptor.java]: Line 124 too.

File: sync-core/src/test/java/com/cloudant/http/HttpTest.java [hunk 3]
```diff
@@ -137,14 +137,14 @@ public void testCookieAuthWithoutRetry() throws IOException {
 
         Assume.assumeFalse(IGNORE_AUTH_HEADERS);
 
-        CookieFilter filter = new CookieFilter(System.getProperty("test.couch.username"),
+        CookieInterceptor filter = new CookieInterceptor(System.getProperty("test.couch.username"),
```
Comment (mikerhodes) [inline sync-core/src/test/java/com/cloudant/http/HttpTest.java]: Var name

File: sync-core/src/test/java/com/cloudant/sync/replication/InterceptorCallCounter.java [hunk 4]
```diff
@@ -14,27 +14,27 @@
...
 /**
- * This request and response filter counts the number of times the filterRequest and filterResponse
+ * This request and response filter counts the number of times the interceptRequest and interceptResponse
  * methods get called.
  */
-public class FilterCallCounter implements HttpConnectionRequestFilter, HttpConnectionResponseFilter {
+public class InterceptorCallCounter implements HttpConnectionRequestInterceptor, HttpConnectionResponseInterceptor {
 
     public int filterRequestTimesCalled = 0;
     public int filterResponseTimesCalled = 0;
```
Comment (mikerhodes) [inline sync-core/src/test/java/com/cloudant/sync/replication/InterceptorCallCounter.java]: Variable names.

File: sync-core/src/test/java/com/cloudant/sync/replication/PullReplicatorTest.java [hunk 5]
```diff
@@ -74,9 +74,9 @@ public void start_StartedThenComplete() throws InterruptedException {
     @Test
     public void testRequestFilters() throws Exception {
 
-        FilterCallCounter filterCallCounter = new FilterCallCounter();
+        InterceptorCallCounter filterCallCounter = new InterceptorCallCounter();
```
Comment (mikerhodes) [inline sync-core/src/test/java/com/cloudant/sync/replication/PullReplicatorTest.java]: variable name

File: sync-core/src/test/java/com/cloudant/sync/replication/PullReplicatorTest.java [hunk 6]
```diff
@@ -85,9 +85,9 @@ public void testRequestFilters() throws Exception {
     @Test
     public void testResponseFilters() throws Exception {
 
-        FilterCallCounter filterCallCounter = new FilterCallCounter();
+        InterceptorCallCounter filterCallCounter = new InterceptorCallCounter();
```
Comment (mikerhodes) [inline sync-core/src/test/java/com/cloudant/sync/replication/PullReplicatorTest.java]: variable name

File: sync-core/src/test/java/com/cloudant/sync/replication/PushReplicatorTest.java [hunk 7]
```diff
@@ -113,10 +113,10 @@ public void testRequestFilters() throws Exception {
 
         //to test the filters we make an invalid request
 
-        FilterCallCounter filterCallCounter = new FilterCallCounter();
+        InterceptorCallCounter filterCallCounter = new InterceptorCallCounter();
```
Comment (mikerhodes) [inline sync-core/src/test/java/com/cloudant/sync/replication/PushReplicatorTest.java]: `testRequestFilters` method name, variable name.

File: sync-core/src/test/java/com/cloudant/sync/replication/PushReplicatorTest.java [hunk 8]
```diff
@@ -125,10 +125,10 @@ public void testRequestFilters() throws Exception {
     @Test
     public void testResponseFilters() throws Exception {
```
Comment (mikerhodes) [inline sync-core/src/test/java/com/cloudant/sync/replication/PushReplicatorTest.java]: method name

File: sync-core/src/test/java/com/cloudant/sync/replication/PushReplicatorTest.java [hunk 9]
```diff
@@ -125,10 +125,10 @@ public void testRequestFilters() throws Exception {
...
 
-        FilterCallCounter filterCallCounter = new FilterCallCounter();
+        InterceptorCallCounter filterCallCounter = new InterceptorCallCounter();
```
Comment (mikerhodes) [inline sync-core/src/test/java/com/cloudant/sync/replication/PushReplicatorTest.java]: variable name

File: sync-core/src/test/java/com/cloudant/sync/replication/PullReplicatorTest.java [hunk 10]
```diff
@@ -74,9 +74,9 @@ public void start_StartedThenComplete() throws InterruptedException {
     @Test
     public void testRequestFilters() throws Exception {
```
Comment (mikerhodes) [inline sync-core/src/test/java/com/cloudant/sync/replication/PullReplicatorTest.java]: method name

File: sync-core/src/test/java/com/cloudant/sync/replication/PullReplicatorTest.java [hunk 11]
```diff
@@ -85,9 +85,9 @@ public void testRequestFilters() throws Exception {
     @Test
     public void testResponseFilters() throws Exception {
```
Comment (mikerhodes) [inline sync-core/src/test/java/com/cloudant/sync/replication/PullReplicatorTest.java]: method name

File: sync-core/src/main/java/com/cloudant/http/CookieInterceptor.java (hunk 2 above)
Comment (rhyshort) [inline sync-core/src/main/java/com/cloudant/http/CookieInterceptor.java]: fixed in a6fa951

File: sync-core/src/test/java/com/cloudant/http/HttpTest.java (hunk 3 above)
Comment (rhyshort) [inline sync-core/src/test/java/com/cloudant/http/HttpTest.java]: fixed in 5392870

File: sync-core/src/test/java/com/cloudant/sync/replication/InterceptorCallCounter.java (hunk 4 above)
Comment (rhyshort) [inline sync-core/src/test/java/com/cloudant/sync/replication/InterceptorCallCounter.java]: done in 9b4ac88

File: sync-core/src/test/java/com/cloudant/sync/replication/PullReplicatorTest.java (hunk 5 above)
Comment (rhyshort) [inline sync-core/src/test/java/com/cloudant/sync/replication/PullReplicatorTest.java]: done in 891faa0

File: sync-core/src/test/java/com/cloudant/sync/replication/PullReplicatorTest.java (hunk 6 above)
Comment (rhyshort) [inline sync-core/src/test/java/com/cloudant/sync/replication/PullReplicatorTest.java]: done in 891faa0

File: sync-core/src/test/java/com/cloudant/sync/replication/PushReplicatorTest.java (hunk 7 above)
Comment (rhyshort) [inline sync-core/src/test/java/com/cloudant/sync/replication/PushReplicatorTest.java]: done in e5fc6b2

File: sync-core/src/test/java/com/cloudant/sync/replication/PushReplicatorTest.java (hunk 8 above)
Comment (rhyshort) [inline sync-core/src/test/java/com/cloudant/sync/replication/PushReplicatorTest.java]: fixed in e5fc6b2

File: sync-core/src/test/java/com/cloudant/sync/replication/PushReplicatorTest.java (hunk 9 above)
Comment (rhyshort) [inline sync-core/src/test/java/com/cloudant/sync/replication/PushReplicatorTest.java]: fixed in e5fc6b2

File: sync-core/src/test/java/com/cloudant/sync/replication/PullReplicatorTest.java (hunk 10 above)
Comment (rhyshort) [inline sync-core/src/test/java/com/cloudant/sync/replication/PullReplicatorTest.java]: done in 1b8892b

File: sync-core/src/test/java/com/cloudant/sync/replication/PullReplicatorTest.java (hunk 11 above)
Comment (rhyshort) [inline sync-core/src/test/java/com/cloudant/sync/replication/PullReplicatorTest.java]: done in 1b8892b

**Comment by mikerhodes** [comment]
I think we're good now. Squash into the first commit, and note in the body of the first commit's comment the reason why we renamed, to avoid the clash with replication filter functions.

Then +1.

**Comment by tomblench** [comment]
+1, did an rgrep for filter and couldn't spot any that you missed

**Pull request merged**